performant-software/neatline-omeka-s
# add JWT authentication system for Neatline API endpoints

### What does this PR do?
- Installs the Firebase JWT library through Composer to support encoding and decoding of trusted user information.
- Adds a new authentication adapter for use with Neatline API calls where the client authenticates by returning a JWT rather than using a key/secret pair.
- Overrides Omeka S's API authentication handling for Neatline endpoints only with the JWT adapter.

### What issues does it address?
- Closes #3 

### How to test
- If currently logged in to https://neatline-3-staging.herokuapp.com/admin, click Logout.
- Go to https://neatline-3-staging.herokuapp.com/s/public-site/neatline . Since you are not logged in to the Omeka S instance, this page should currently show only Neatline exhibits which have been set as public.
- Log in to https://neatline-3-staging.herokuapp.com/admin (will send credentials over chat).
- Return to https://neatline-3-staging.herokuapp.com/s/public-site/neatline . Since you are logged in as an admin, you should now be able to see at least one additional exhibit that has not been set as public in the list of Neatline exhibits.
